arch/sim/src/sim/up_linuxi2c.c

```diff
@@ -0,0 +1,145 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/kmalloc.h>
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <nuttx/i2c/i2c_master.h>
+
+#include "up_linuxi2c_host.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int linuxi2c_transfer(FAR struct i2c_master_s *dev,
+                             FAR struct i2c_msg_s *msgs, int count);
+#ifdef CONFIG_I2C_RESET
+int linuxi2c_reset(FAR struct i2c_master_s *dev);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+struct i2c_ops_s g_linuxi2c_ops =
+{
+  .transfer = linuxi2c_transfer,
+#ifdef CONFIG_I2C_RESET
+  .reset = linuxi2c_reset,
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int linuxi2c_transfer(FAR struct i2c_master_s *dev,
+                             FAR struct i2c_msg_s *msgs, int count)
+{
+  int ret;
+  uint16_t msg_idx;
+  struct linuxi2c_master_s *priv = (struct linuxi2c_master_s *)dev;
+  struct linux_i2c_rdwr_ioctl_data ioctl_data;
+  struct linux_i2c_msg *l_msgs =
+    kmm_malloc(sizeof(struct linux_i2c_msg) * count);
+
+  if (l_msgs == NULL)
+    {
+      return -1;
+    }
+
+  ioctl_data.msgs = l_msgs;
+  ioctl_data.nmsgs = count;
+
+  for (msg_idx = 0; msg_idx < count; msg_idx++)
+    {
+      /* TODO: Translate flags */
+    }
+
+  ret = linuxi2c_transfer_host(priv, &ioctl_data);
+  kmm_free(l_msgs);
+  return ret;
+}
+
+#ifdef CONFIG_I2C_RESET
+static int linuxi2c_reset(FAR struct i2c_master_s *dev)
+{
+  return -1;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: linuxi2c_initialize
```

Comment:
it's better to use a generic name here since only the host side need to distinguish Linux, Windows, macOS.